cloud/scope/machine_pool.go

```diff
@@ -472,7 +520,10 @@ func (m *MachinePoolScope) UpdatePool(ctx context.Context, instancePool *core.In
 		}
 
 		req := core.UpdateInstancePoolRequest{InstancePoolId: instancePool.Id,
-			UpdateInstancePoolDetails: core.UpdateInstancePoolDetails{Size: common.Int(replicas)}}
+			UpdateInstancePoolDetails: core.UpdateInstancePoolDetails{
+				Size:                    common.Int(replicas),
+				InstanceConfigurationId: common.String(m.OCIMachinePool.Spec.InstanceConfiguration.InstanceConfigurationId),
```

Comment:
So this will work, but doesn't really work great given this won't cordon/drain each node and then delete and recreate. I'm thinking this should be pulled from this PR then create a new issue to specifically address the upgrade issue. What do you think @shyamradhakrishnan

Reply:
we have to first create a new instance configuration and update that into the instance pool if there is an update right? I think we should do that, I mean do the update but no drain logic. We should slo most probably need to update the cloud init.

Reply:
https://github.com/oracle/cluster-api-provider-oci/issues/95